### PR TITLE
Fix labels in anonymous mode

### LINKED ISSUE #4305: [BUG] anonym. mode: my labels error
**To Reproduce**
1. use FreshRSS in anonym. mode
2. open an article
3. click on "my labels"
4. see browser console (error displayed)
5. see the empty label list

grafik

**Expected behavior**

- No error
- show added labels

**Environment information (please complete the following information):**
- Browser: Firefox 98
- FreshRSS version: 1.19.2 (see demo.freshrss.org), 1.20.0-dev

### PR DESCRIPTION
**Pull request by Alkarex**
fix https://github.com/FreshRSS/FreshRSS/issues/4305
Replaces https://github.com/FreshRSS/FreshRSS/pull/4649

### CODE REVIEW

File: p/scripts/main.js [hunk 1]
```diff
@@ -1300,46 +1300,53 @@ function loadDynamicTags(div) {
...
 		let html = '';
 		if (json && json.length) {
 			for (let i = 0; i < json.length; i++) {
 				const tag = json[i];
+				if (context.anonymous && !tag.checked) {
+					// In anomymous mode, show only the used tags
+					continue;
+				}
 				html += '<li class="item"><label><input class="checkboxTag" name="t_' + tag.id + '" type="checkbox"' +
+						(context.anonymous ? ' disabled="disabled"' : '') +
```
Comment (math-GH) [inline p/scripts/main.js]: Please have a look into line 1201. The event listener does not make sense in context.anonymous = true on the checkbox.

**Comment by math-GH** [comment]
If there is no label assigned then the popup list looks a bit broken:

grafik
The title is given but nothing else

File: p/scripts/main.js (hunk 1 above)
Comment (Alkarex) [reply p/scripts/main.js]: The `if (checkboxTag) {` should do the job, doesn't it? In anonymous mode, this is false

Comment (math-GH) [reply p/scripts/main.js]: `if (ev.target.closest('.checkboxTag'))` is independend of context nor `disabled="disabled"`

**Comment by Alkarex** [comment]
> If there is no label assigned then the popup list looks a bit broken:

What about now?

**Comment by math-GH** [comment]
> > If there is no label assigned then the popup list looks a bit broken:
> 
> What about now?

IMHO it does not make sense to list all (incl. not used) labels in the anonym. mode

grafik


I would see follwing 2 options in case of that there is no label added:
- hide the "label" link (I guess that it would need some additional SQL statement)
- give the information "no label set" in the pop-up

File: p/scripts/main.js (hunk 1 above)
Comment (Alkarex) [reply p/scripts/main.js]: Indeed, but there is no `.checkboxTag` in anonymous mode, so the rest is just skipped as it should, isn't it ?

**Comment by Alkarex** [comment]
I have added a message when there are no labels.
Maybe it could be styled a bit better if you feel like it @math-GH 

image

I would like to keep the labels also in anonymous mode, as there are more features coming related to labels (such as automatic labelling).

File: p/scripts/main.js (hunk 1 above)
Comment (math-GH) [reply p/scripts/main.js]: The class is still there 

grafik

grafik

**Comment by math-GH** [comment]
No labels style improved.

File: p/scripts/main.js (hunk 1 above)
Comment (Alkarex) [reply p/scripts/main.js]: Sorry, my mistake. Better now?

**Comment by math-GH** [comment]
It looks fine now.

Does it make sense to keep the background color while hovering in this situation? There is no click action behind. As a user I would expect something there

grafik

As far as I can oversee it there is no CSS class for the anonym. mode, isn't it?

**Comment by Alkarex** [comment]
> Does it make sense to keep the background color while hovering

I do not mind much, but I find it fine as it is, and I would rather limit the amount of code, especially when specific to a relatively niche feature

**Comment by Frenzie** [comment]
Fwiw you might be able to do something with the new :has() selector, but wouldn't it make more sense to simply not show the checkbox instead of disabling it?

**Pull request merged**